server/data/incidentTypeConfiguration/formWizard.ts

```diff
@@ -127,12 +113,53 @@ export function generateFields(config: IncidentTypeConfiguration): FormWizard.Fi
   return fields
 }
 
+/**
+ * Returns the list of next step conditions for the question/answers
+ *
+ * Answers leading to the same question are grouped together.
+ *
+ * @param question
+ * @param answers
+ * @returns next step conditions
+ */
+function nextSteps(question: QuestionConfiguration, answers: AnswerConfiguration[]): FormWizard.NextStepCondition[] {
+  // Group answers by next question id
+  const groupedByNextQuestion: Map<string | null, AnswerConfiguration[]> = new Map()
```

Comment:
I was using a plain object `{}` here but tests highlighted the fact that when then iterating over its keys/values the key was converted to `string`.

Beside from being surprising this is bad in our case because a `nextQuestionId` (used to group answers) **can** be `null` and we don't want this to be converted to `'null'` which would be incorrectly interpreted as "go to the question with id 'null'" which is wrong.

Side rant: this is also why below there is a for loop instead of using `map()` on the return value of `.entries()`: That value doesn't have a `map()` method (unlike `Object.entries()` which can be mapped)

Reply:
we get [iterator helpers](https://github.com/tc39/proposal-iterator-helpers) in node 22; which goes LTS next week 💪